executors/http/src/main/java/net/opentsdb/query/execution/HttpQueryV2Executor.java

```diff
@@ -398,11 +398,11 @@ public void execute() {
       final HttpPost post = new HttpPost(endpoint);
       final Object session_headers = context.getSessionObject(
           SESSION_HEADERS_KEY);
-      if (!(session_headers instanceof Map)) {
-        throw new IllegalStateException("Session headers were not a map: " 
-            + session_headers.getClass());
-      }
       if (session_headers != null) {
+        if (!(session_headers instanceof Map)) {
+          throw new IllegalStateException("Session headers were not a map: "
```

Comment:
Ah do something similar to this instead of throwing:
```
final Exception ex = new RejectedExecutionException(
  "Failed to convert query", e);
callback(ex, TsdbTrace.exceptionTags(ex));
```

That way it'll bubble up.